jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -159,47 +229,126 @@ def _on_spectral_unit_changed(self, *args):
+                if check_if_unit_is_per_solid_angle(current_y):
+                    self.flux_or_sb.selected = 'Flux'
 
+                untranslatable_units = self.untranslatable_units()
+                # disable translator if flux unit is untranslatable,
+                # still can convert flux units, this just disables flux
+                # to surface brightnes translation for units in list.
+                if flux_or_sb in untranslatable_units:
+                    self.can_translate = False
+                else:
+                    self.can_translate = True
+
+            elif arg['name'] == 'sb_unit_selected':
+                if data_collection_unit == 'Flux':
+                    # when the configuration is Specviz, translation is not currently supported.
+                    # If in Cubeviz, all spectra pass through Spectral Xxtraction plugin and will
+                    # have a scale factor assigned in the metadata, enabling translation.
+                    raise ValueError(
+                        "Unit translation between Flux and Surface Brightness "
+                        "is not supported in Specviz."
+                    )
+                flux_or_sb = self.sb_unit.selected
+                self.can_translate = True
+                # update flux or surface brightness dropdown if necessary
+                if not check_if_unit_is_per_solid_angle(current_y):
+                    self.flux_or_sb.selected = 'Surface Brightness'
+            else:
+                return
+
+        yunit = _valid_glue_display_unit(flux_or_sb, self.spectrum_viewer, 'y')
+
+        # on instantiation, we set determine flux choices and selection
+        # after surface brightness
+        if not self.flux_unit.choices:
+            return
+        if self.spectrum_viewer.state.y_display_unit != yunit:
             self.spectrum_viewer.state.y_display_unit = yunit
             self.hub.broadcast(GlobalDisplayUnitChanged('flux',
-                                                        self.flux_or_sb_unit.selected,
+                                                        flux_or_sb,
                                                         sender=self))
+            self.spectrum_viewer.reset_limits()
 
-    # Ensure first dropdown selection for Flux/Surface Brightness
-    # is in accordance with the data collection item's units.
-    @observe('show_translator')
-    def _set_flux_or_sb(self, *args):
-        if (self.spectrum_viewer and hasattr(self.spectrum_viewer.state, 'y_display_unit')
-           and self.spectrum_viewer.state.y_display_unit is not None):
-            if u.sr in u.Unit(self.spectrum_viewer.state.y_display_unit).bases:
-                self.flux_or_sb_selected = 'Surface Brightness'
-            else:
-                self.flux_or_sb_selected = 'Flux'
-
-    @observe('flux_or_sb_selected')
+    @observe('flux_or_sb_selected', 'flux_unit_selected', 'sb_unit_selected')
```

Comment:
there are two methods (this and `_on_flux_unit_changed`) that observe `flux_unit_selected` and `sb_unit_selected`.  Does order matter?  Or can one just call the other so that we ensure a consistent order?

Reply:
Before your commit, we didn't broadcast the GlobalDisplayUnitChanged message in the translator, so order did matter before then (where `translate_units` would be before `_on_flux_unit_changed`). I will double check this now and leave a follow up comment.

There is the case where you would enter _translate on a unit conversion (non-translation) and leave it before actually translating (since the observe would trigger for `flux_unit_selected `and `sb_unit_selected`). You'd still want to trigger `_on_flux_unit_changed`.

Let me consider more if we put a call to `_translator` into `_on_flux_unit_changed`. I think it's doable and might strip out some of the duplicated logic.

Reply:
I can't produce a case where the order seems to matter. I am working on calling `_translator` into `_on_flux_unit_changed` right now, and at the moment I think all the other comments have been address @kecnry

Reply:
`_on_flux_unit_changed` now calls `_translator`, and since `_translator` doesn't have the observes, I combined the two separate translator functions into one @kecnry